### PR TITLE
feat(list): add fixed-size list support for value_counts

### LINKED ISSUE #3519: `Expression.list.value_counts` does not work for fixed size lists
### Describe the bug

Type checking doesn't allow the user to pass in a fixed size list. But even after removing that, the internal implementation for the expression doesn't work for fixed size lists.

### To Reproduce

e.g.
```
table = MicroPartition.from_pydict(
    {
        # Test list of an atomic type.
        "col1": [["a"], ["ab", "a"], [None, "a", "", None, "a", "b", "c"], None, ["a", ""]],
        # Test lists of a nested type.
        "col2": [
            [[1]],
            [[3, 3], [4], [5, 5]],
            [],
            [[], []],
            None,
        ],
        "start": [-1, 1, 0, 2, -2],
        "end": [1, 2, 0, 4, 3],
        "edge_start": [-1, -2, -5, 0, -2],
        "edge_end": [-1, -2, -1, -2, -1],
    }
)
result = table.eval_expression_list(
    [
        col("col1").list.distinct().alias("col1"),
    ]
)
print(f"result {result}")
```

### Expected behavior

_No response_

### Component(s)

Expressions

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by andrewgazelka**
Closes #3519

### CODE REVIEW

**Comment by andrewgazelka** [comment]
* **#3521** <a href="https://app.graphite.dev/github/pr/Eventual-Inc/Daft/3521?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Eventual-Inc/Daft/3521?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by andrewgazelka** [comment]
also unsure whether this should be feat or fix... given to end user it might seem like a feat? 😅

File: src/daft-core/src/array/fixed_size_list_array.rs [hunk 1]
```diff
@@ -198,8 +198,18 @@ impl FixedSizeListArray {
     }
 
     pub fn to_list(&self) -> ListArray {
+        let field = self.field.clone();
+
+        let DataType::FixedSizeList(inner_type, _) = &field.dtype else {
+            panic!("Expected FixedSizeListArray, got {:?}", field.dtype);
```
Comment (universalmind303) [inline src/daft-core/src/array/fixed_size_list_array.rs:204]: I think this should be `unreachable!` as this should be infallible at this point.

File: src/daft-core/src/array/fixed_size_list_array.rs [hunk 2]
```diff
@@ -198,8 +198,18 @@ impl FixedSizeListArray {
...
+        };
+
+        let datatype = DataType::List(inner_type.clone());
+        let mut field = (*field).clone();
+        field.dtype = datatype;
+
```
Comment (universalmind303) [inline src/daft-core/src/array/fixed_size_list_array.rs:210]: why are we cloning the field twice. once on 201, and again here.

**Review by jaychia — COMMENTED** [review]
LGTM, agreed with  @universalmind303's comments will let him do the final signoff

File: src/daft-core/src/array/fixed_size_list_array.rs (hunk 1 above)
Comment (andrewgazelka) [reply src/daft-core/src/array/fixed_size_list_array.rs]: 1a2ba6f496c4ad2e0a1fbf7262e07b249a4ad117

File: src/daft-core/src/array/fixed_size_list_array.rs (hunk 2 above)
Comment (andrewgazelka) [reply src/daft-core/src/array/fixed_size_list_array.rs:210]: 1a2ba6f496c4ad2e0a1fbf7262e07b249a4ad117

**Review by universalmind303 — APPROVED** [review]

**Pull request merged**